semantic_release/hvcs.py

```diff
@@ -59,6 +60,26 @@ def _fix_mime_types():
     mimetypes.add_type("text/markdown", ".md")
 
 
+class TokenAuth(AuthBase):
```

Comment:
Using a request `AuthBase` implementation make things easier to handle, as the internal requests netrc Auth is used only if no auth is configured on the request.

In previous implementation, `Authorization` header is set manually and auth is not set. So this header is replaced by internal netrc Auth implementation, making `.netrc` file having a higher priority than `GH_TOKEN` environment variable.

With this change, `GH_TOKEN` will always have the priority if defined, even if `.netrc` file is available on the host.